torch/csrc/cuda/comm.cpp

```diff
@@ -111,4 +118,89 @@ tensor_list2d broadcast_coalesced(TensorList tensors, IntList devices, size_t bu
   return outputs;
 }
 
-}}
+std::vector<at::Tensor> scatter(
+    const at::Tensor& tensor,
+    at::IntList devices,
+    const at::optional<std::vector<int64_t>>& chunk_sizes,
+    int64_t dim,
+    const at::optional<std::vector<THCStream*>>& streams) {
+  std::vector<at::Tensor> chunks;
+  if (!chunk_sizes) {
+    chunks = tensor.chunk(/*chunks=*/devices.size(), /*dim=*/dim);
+  } else {
+    const int64_t chunk_size_sum =
+        std::accumulate(chunk_sizes->begin(), chunk_sizes->end(), 0);
+    // clang-format off
+    AT_CHECK(
+      chunk_size_sum == tensor.size(dim),
+      "given chunk sizes don't sum up to the tensor's size ",
+      "(sum(chunk_sizes) == ", chunk_size_sum,
+      ", but expected ", tensor.size(dim), ")");
+    // clang-format on
+    chunks.reserve(chunk_sizes->size());
+    int64_t chunk_start = 0;
+    for (size_t chunk = 0; chunk < chunk_sizes->size(); ++chunk) {
+      const int64_t chunk_size = (*chunk_sizes)[chunk];
+      AT_CHECK(chunk_size > 0, "Chunk size must be positive");
+      chunks.push_back(tensor.narrow(dim, chunk_start, chunk_size));
+      chunk_start += chunk_size;
+    }
+    AT_ASSERT(chunks.size() == chunk_sizes->size());
```

Comment:
Well this one is quite easy to prove yourself (you `push_back` at every iteration, there are `chunk_sizes->size()` iters because you never beak out of the loop), but we can leave it if you really want.